File: Results/DER/Translation/translation_output/w_test/granite/codenet/python/java/p02792.java
System:
You are an expert java programmer and assistant

Question:
Translate the following code from python to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
25

Expected output:
17


```
N=int(raw_input())

num = [[0 for i in range(10)] for j in range(10)]

for i in range(1,N+1):
    if i % 10 > 0:
        num[int(str(i)[0])][i % 10] += 1

#print num

c = 0
for i in range(1,10):
    for j in range(1,10):
        c += num[i][j]*num[j][i]
print c

```


Answer:
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[][] num = new int[10][10];

        for (int i = 1; i <= N; i++) {
            if (i % 10 > 0) {
                num[Integer.parseInt(String.valueOf(i).charAt(0) + "")]
                        [i % 10] += 1;
            }
        }

        int c = 0;
        for (int i = 1; i < 10; i++) {
            for (int j = 1; j < 10; j++) {
                c += num[i][j] * num[j][i];
            }
        }
        System.out.println(c);
    }
}
```

Test input:
25

Expected output:
17<|endoftext|>

File: Results/DER/Translation/translation_output/w_test/granite/codenet/java/python/p02792.py
System:
You are an expert python programmer and assistant

Question:
Translate the following code from java to python and enclose your solution inside ```python```.
A sample test case is provided below:

Test input:
25

Expected output:
17


```
import java.util.*;
import static java.lang.Math.*;
import java.math.BigInteger;

public class p02792 {
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		
		// å¥å
		int n = sc.nextInt();
		
		// è¨ç®
		int result = 0;
		int[][] array = new int[10][10];
		for(int i = 1; i <= n; i++){
		    int head = Integer.valueOf(String.valueOf(i).substring(0,1));
		    int tail = i%10;
		    array[head][tail]++;
		}
		
		int keta = String.valueOf(n).length();
		for(int i = 1; i <= n; i++){
		    int head = Integer.valueOf(String.valueOf(i).substring(0,1));
		    int tail = i%10;
		    if(tail == 0) continue;
		    result += array[tail][head];
		}
		
		// åºå
		System.out.println(result);
	}
}



```


Answer:
```python
import sys

def main():
    n = int(sys.stdin.readline())
    count = 0
    table = [[0 for _ in range(10)] for _ in range(10)]
    for i in range(1, n + 1):
        head = int(str(i)[0])
        tail = i % 10
        table[head][tail] += 1
    for i in range(1, n + 1):
        head = int(str(i)[0])
        tail = i % 10
        if tail == 0:
            continue
        count += table[tail][head]
    print(count)

if __name__ == '__main__':
    main()
```


Test input:
```
25
```

Expected output:
```
17
```<|endoftext|>
